keras_nlp/layers/transformer_encoder.py

```diff
@@ -114,53 +114,55 @@ def _build(self, input_shape):
         # Create layers based on input shape.
         self._built = True
         self._input_shape = input_shape
-        feature_size = input_shape[-1]
-        self._attention_head_size = int(feature_size // self.num_heads)
-        self._multi_head_attention_layer = keras.layers.MultiHeadAttention(
+        # Infer the dimension of our hidden feature size from the build shape.
+        hidden_dim = input_shape[-1]
+        # Attention head size is `hidden_dim` over the number of heads.
+        head_dim = int(hidden_dim // self.num_heads)
```

Comment:
Why not call this `key_dim`? The heads have vectors of many dimensions so `head_dim` doesn't seem very descriptive

Reply:
sgtm